### PR TITLE
Fix #4777: Datatable typescript improvements

### LINKED ISSUE #4777: DataTable: Type issue for Selection Mode
In single selection mode, the type should represent single value, whereas in multiple selection mode, it should represent several values.

`type DataTableSelection<TValue extends DataTableValueArray> = DataTableRowData<TValue> | DataTableRowDataArray<TValue> | DataTableCellSelection<TValue>;`

### PR DESCRIPTION
**Pull request by melloware**
Fix #4777: Datatable typescript improvements

### CODE REVIEW

File: components/lib/datatable/datatable.d.ts [hunk 1]
```diff
@@ -1558,6 +1585,71 @@ export interface DataTableProps<TValue extends DataTableValueArray> extends Omit
...
+     * Whether to cell selection is enabled or not.
+     * @defaultValue false
+     */
+    cellSelection?: false | undefined;
+    /**
+     * Specifies the selection mode, valid values are "single", "multiple", "radiobutton" and "checkbox".
+     */
+    selectionMode?: 'single' | 'radiobutton' | undefined;
+    /**
+     * Selected row in single mode or an array of values in multiple mode.
```
Comment (bu3alwa) [inline components/lib/datatable/datatable.d.ts:1599]: The comments need to be updated if we have a different type of single and multiple now

File: components/doc/datatable/rowselection/singledoc.js [hunk 2]
```diff
@@ -67,16 +67,16 @@ import { InputSwitch, InputSwitchChangeEvent } from 'primereact/inputswitch';
...
-    code: string;
-    name: string;
-    description: string;
-    image: string;
-    price: number;
-    category: string;
-    quantity: number;
-    inventoryStatus: string;
-    rating: number;
+    id?: string;
```
Comment (bu3alwa) [inline components/doc/datatable/rowselection/singledoc.js:70]: Why are these all possibly undefined? Doesn't the response from the data provide that object without any missing fields?

Comment (melloware) [reply components/doc/datatable/rowselection/singledoc.js:70]: I always copy and paste this object around while testing so I made them all optional to make my life easier when testing 😄

**Review by bu3alwa — APPROVED** [review]

File: components/doc/datatable/pt/ptdoc.js [hunk 3]
```diff
@@ -192,7 +192,6 @@ export default function PTDemo() {
                 }}
             ></Column>
         </DataTable>
-    </div>
```
Comment (melloware) [inline components/doc/datatable/pt/ptdoc.js:195]: This was a renegade DIV breaking the Passthrough stackblitz.

**Pull request merged**